test/e2e/azure_machinepool_drain.go

```diff
@@ -176,35 +180,33 @@ func testMachinePoolCordonAndDrain(ctx context.Context, mgmtClusterProxy, worklo
 		}
 
 		return errors.New("no machine has started to drain")
-	})
-
-	By(fmt.Sprintf("checking for a machine to successfully complete draining for machine pool: %s/%s", amp.Namespace, amp.Name))
-	Eventually(func() error {
-		ampmls, err := getAzureMachinePoolMachines(ctx, mgmtClusterProxy, workloadClusterProxy, amp)
-		if err != nil {
-			return errors.Wrap(err, "failed to list the azure machine pool machines")
-		}
+	}, waitForDrainOperationTimeout, waitForDrainSleepBetweenRetries).Should(Succeed())
 
-		for _, machine := range ampmls {
-			if conditions.Has(&machine, clusterv1.DrainingSucceededCondition) && conditions.IsTrue(&machine, clusterv1.DrainingSucceededCondition) {
-				return nil // started draining the node prior to delete
-			}
-		}
-
-		return errors.New("no machine has finished draining")
-	})
+	// TODO setup a watcher to detect the terminal drain success state
```

Comment:
Rather than checking for the expected state transitions serially, we can improve this in the future by setting a watch on the machine in the process of deleting. I'm making the judgment that doing that work is out of scope for this PR (this PR is about standardizing the usage of Eventually() for retryable operations).

Bottom line, this test coverage was never actually running due to the lack of `.Should(Succeed())`, so adding the "validate that drain begins after delete" is already net additive coverage.

Reply:
can we open an issue for this rather than leaving a TODO in the code? Also does this mean #2120 isn't fully fixed?

Reply:
The referenced issue documents a flake that should be addressed w/ the new `Eventually()` block in L191 below